src/kudu/consensus/log.cc

```diff
@@ -454,15 +454,21 @@ Status Log::Open(const LogOptions &options,
   RETURN_NOT_OK(env_util::CreateDirIfMissing(
       fs_manager->env(), tablet_wal_path));
 
-  scoped_refptr<Log> new_log(new Log(options,
-                                     fs_manager,
-                                     tablet_wal_path,
-                                     tablet_id,
+  scoped_refptr<Log> new_log;
+  if (options.log_factory) {
+    RETURN_NOT_OK(options.log_factory->createLog(options,
+        fs_manager, tablet_wal_path,
+        tablet_id, metric_entity, &new_log));
+  } else {
```

Comment:
When would `log_factory` not be initialized?

Reply:
when tablet server options has not used a factory class, for example a future application which just wants to use kuduraft log AS-IS